ksql-functional-tests/src/test/resources/query-validation-tests/udf-tests.json

```diff
@@ -0,0 +1,119 @@
+        {"topic": "OUTPUT", "value": {"R0" : 2, "R00" : 2.0, "R1" :  1.7, "R2" : 1.7,  "R10" :  1.7, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 2, "R00" : 2.0, "R1" :  1.5, "R2" : 1.5,  "R10" :  1.5, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 3, "R00" : 3.0, "R1" :  3.0, "R2" : 3.0,  "R10" :  3.0, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 1, "R00" : 1.0, "R1" :  1.2, "R2" : 1.23, "R10" :  1.234567,"1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 0, "R00" : 0.0, "R1" :  0.0, "R2" : 0.0,  "R10" :  0.0, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 111, "R00" : 111.0, "R1" :  111.0, "R2" : 111.0,  "R10" :  111.0, "1R" :  110.0 ,"2R" :  100.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : null, "R00" : null, "R1" :  null, "R2" : null, "R10" : null,"1R" :  null ,"2R" :  null }},
+        {"topic": "OUTPUT", "value": {"R0" : -1, "R00" : -1.0, "R1" :  -1.1, "R2" : -1.1,  "R10" :  -1.1, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -1, "R00" : -1.0, "R1" :  -1.5, "R2" : -1.5,  "R10" :  -1.5, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -2, "R00" : -2.0, "R1" :  -1.7, "R2" : -1.7,  "R10" :  -1.7, "1R" :  0.0 ,"2R" :  0.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : 12346, "R00" : 12346.0, "R1" :  12345.7, "R2" : 12345.68,  "R10" :  12345.678, "1R" :  12350.0 ,"2R" :  12300.0 }},
+        {"topic": "OUTPUT", "value": {"R0" : -12346, "R00" : -12346.0, "R1" :  -12345.7, "R2" : -12345.68,  "R10" :  -12345.678, "1R" :  -12350.0 ,"2R" :  -12300.0 }}
+      ]
+    },
+    {
+      "name": "calculate ROUND with large DECIMAL values",
+      "statements": [
+        "CREATE STREAM test (v DECIMAL(33, 16)) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE STREAM OUTPUT AS SELECT ROUND(v) as R0, ROUND(v, 0) as R00, ROUND(v, 1) as R1, ROUND(v, 2) as R2, ROUND(v, 10) as R10, ROUND(v , -1) as 1R , ROUND(v , -2) as 2R  FROM test;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": "12345678987654321.2345678987654321"}
+      ],
+      "outputs": [
+        {
+          "topic": "OUTPUT",
+          "value": "\"12,345,678,987,654,321\",\"12,345,678,987,654,321.0000000000000000\",\"12,345,678,987,654,321.2000000000000000\",\"12,345,678,987,654,321.2300000000000000\",\"12,345,678,987,654,321.2345678988000000\",\"12,345,678,987,654,320.0000000000000000\",\"12,345,678,987,654,300.0000000000000000\""
+        }
+      ],
+      "post": {
+        "sources": [
+          {
+            "name": "OUTPUT",
+            "type": "stream",
+            "valueSchema": "STRUCT<R0 DECIMAL(17,0), R00 DECIMAL(33,16), R1 DECIMAL(33,16), R2 DECIMAL(33,16), R10 DECIMAL(33,16), 1R DECIMAL(33,16), 2R DECIMAL(33,16)>"
+          }
+        ]
+      }
+    },
+    {
+      "name": "calculate FLOOR function",
```

Comment:
Does this duplicate the existing test for Floor in `math.json`?

Reply:
Yes, I've improved the test implementation